django/http/request.py

```diff
@@ -278,28 +282,37 @@ def _load_post_and_files(self):
             self._mark_post_parse_error()
             return
 
-        if self.META.get('CONTENT_TYPE', '').startswith('multipart/form-data'):
-            if hasattr(self, '_body'):
-                # Use already read data
-                data = BytesIO(self._body)
-            else:
-                data = self
-            try:
+        try:
+            if self.META.get('CONTENT_TYPE', '').startswith('multipart/form-data'):
+                if hasattr(self, '_body'):
+                    # Use already read data
+                    data = BytesIO(self._body)
+                else:
+                    data = self
```

Comment:
That section specifically does not need to be inside the "try" but in order to capture all possible sections that raise the exception it was included.